src/neural/cuda/network_cudnn.cc

```diff
@@ -1078,6 +1078,15 @@ std::unique_ptr<Network> MakeCudnnNetwork(const std::optional<WeightsFile>& w,
           pblczero::NetworkFormat::NETWORK_CLASSICAL_WITH_HEADFORMAT &&
       weights.format().network_format().network() !=
           pblczero::NetworkFormat::NETWORK_SE_WITH_HEADFORMAT) {
+    // Check if we must switch to cuda.
+    if (weights.format().network_format().network() ==
```

Comment:
Maybe move this `if` outside of the outer `if` as if the inner is true, outer will also be true?

Also, switch may be more readable (not sure), something like:

```cpp
switch (weights.format().network_format().network()) {
  case pblczero::NetworkFormat::NETWORK_CLASSICAL_WITH_HEADFORMAT:
  case pblczero::NetworkFormat::NETWORK_SE_WITH_HEADFORMAT:
      break;
  case pblczero::NetworkFormat::NETWORK_ATTENTIONBODY_WITH_HEADFORMAT:
      CERR << "Nework format not supported directly, switching backend.";
      return NetworkFactory::Get()->Create(
          "cuda" +
              std::string(std::is_same<half, DataType>::value ? "-fp16" : ""),
          weights, options);
   default:
      throw Exception("Network format " +
                    pblczero::NetworkFormat::NetworkStructure_Name(
                        weights.format().network_format().network()) +
                    " is not supported by CuDNN backend.");
}
```

Reply:
I liked the switch version better.